client/signup/steps/site-info-collection/accordion-form.tsx

```diff
@@ -1,36 +1,77 @@
-import AccordionFormSection from './accordion-form-section';
-import { AccordionSectionProps, ValidatorFunction } from './types';
+import React, { ReactElement, useState, JSXElementConstructor, useEffect } from 'react';
 
 interface AccordionFormProps {
-	sections: AccordionSectionProps[];
-	currentIndex: number;
-	isSectionAtIndexTouched: Record< string, boolean >;
-	onNext: ( validator?: ValidatorFunction ) => void;
-	onOpen: ( currentIndex: number ) => void;
+	initialOpenSectionId: string;
+	isGoToNextAllowed: ( sectionID: string ) => boolean;
+	onOpenCallback?: ( openedSectionId: string ) => void;
+	onNextCallback?: ( openedSectionId: string ) => void;
+	onSubmit: () => void;
+	children: ReactElement< any, string | JSXElementConstructor< any > >[];
+}
+
+function getChildrenIds( elements: ReactElement< any, string | JSXElementConstructor< any > >[] ) {
+	const eArray = React.Children.toArray( elements );
+
+	const allIds = eArray.map( ( element ) => {
+		return element.props.sectionId;
+	} );
+	return allIds;
 }
 
 export function AccordionForm( {
-	sections,
-	currentIndex,
-	isSectionAtIndexTouched,
-	onNext,
-	onOpen,
+	children,
+	onNextCallback,
+	onSubmit,
+	onOpenCallback,
+	initialOpenSectionId,
+	isGoToNextAllowed,
 }: AccordionFormProps ) {
+	const [ currentOpenSectionId, setCurrentOpenSectionId ] = useState< string >(
+		initialOpenSectionId
+	);
```

Comment:
Section id is now a string based Id. The logic for `next` is handled by figuring out the indexOf the sectionId and then incrementing and setting that.